awx/main/tests/functional/commands/test_callback_receiver.py

```diff
@@ -24,3 +32,108 @@ def test_wrapup_does_send_notifications(mocker):
     job.refresh_from_db()
     assert job.host_status_counts == {}
     mock.assert_called_once_with('succeeded')
+
+
+class FakeRedis:
+    def keys(self, *args, **kwargs):
+        return []
+
+    def set(self):
+        pass
+
+    def get(self):
+        return None
+
+    @classmethod
+    def from_url(cls, *args, **kwargs):
+        return cls()
+
+    def pipeline(self):
+        return self
+
+
+class TestCallbackBrokerWorker(TransactionTestCase):
```

Comment:
Background I maybe should have put as comments - the callback receiver code makes copious use of catching database errors, and then taking followup actions. This does not play well with transactions, because you get the error that you are in a broken transactions so you can't make subsequent queries. That error handling is the core behavior that's being tested here, so I can't workaround other than running the test in auto-commit mode. I get that with the `TransactionTestCase`.

Not quite sure what the collateral damage is, although I suspect that it affects what `.sqlite3` files are created.